ls1intum/Artemis
# `Communication`: Migrate posts to course-wide channels

## Do not deploy on test server, contains migration



### Checklist
#### General



### Motivation and Context


With #7437 posts are no longer shown in the communication tab. Therefore, they are migrated to the new course-wide channels

### Description

Legacy lecture, exercise, and course-wide-topic posts are migrated to the corresponding course-wide channels.
If channels for lectures, exercises, or course-wide topics do not exist for a course, they are created. Posts are linked with the corresponding channel. Finally, the reference from posts to a lecture/exercise/course-wide-topic is set to null, so that they only have a reference to a conversation.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with communication and messaging enabled

1. Do not deploy yet! 
2. Delete (some) general course-wide, lecture, and exercise channels from a course
3. Choose long names/names with special characters/... for lectures/exercises where you deleted a channel
4. Deactivate messaging in the course
5. Create course-wide posts on the communication tab
6. Create lecture/exercise posts on the corresponding lecture/exercise details page for lectures/exercises where you deleted channels
7. Deploy the migration
8. Activate messaging in the course
9. Check that all channels have been created and that posts (which are now messages) are shown in the channels. No posts should be shown in the communication tab anymore


### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage






Same as develop

### Screenshots


No UI changes
